### PR TITLE
Multi-arch build

### PR DESCRIPTION
**Pull request by donporter**
Let's try adopting the hooks from here: https://github.com/Tob1as/docker-build-example

Looks like docker hub doesn't do multi-arch by default, but there are sufficient hooks to cajole it into a multi-arch build.

### CODE REVIEW

**Comment by donporter** [comment]
Assuming we want to deploy this, this PR is ready to merge (once reviewed).

File: hooks/pre_build
Comment (eric-unc) [inline hooks/pre_build:24]: Do we need all of these comments for latter, or is it just debug stuff?

**Comment by eric-unc** [comment]
I don't know a lot about Docker image building but I don't see anything broadly wrong?

**Review by eric-unc — APPROVED** [review]

File: hooks/pre_build
Comment (donporter) [reply hooks/pre_build:24]: I think some of these are outlining what the actual command does.  I ended up leaving them there because they helped me understand what the uncommented lines are doing.

**Comment by donporter** [comment]
Let us not merge this until after LDOC, per out-of-band conversation.  The multiarch image can be built from this branch without merging.

I also think this failed docker ci job is spurious - the builds in dockerhub all passed, but the service that posts status is offline.

**Pull request merged**
